src/workflow-handler.ts

```diff
@@ -171,7 +147,7 @@ export class WorkflowHandler {
     return runs[0].id as number;
   }
 
-  private async findWorklowRunIdFromRunName(runName: string): Promise<number> {
+  private async findWorklowRunFromRunName(runName: string): Promise<WorkflowRunResult> {
     const result = await this.octokit.rest.checks.listForRef({
       check_name: runName,
```

Comment:
I tested your branch @evanlindsey and noticed it was still not working for me.

Having a deeper look, I figure out that current function `octokit.rest.checks.listForRef`, which is based on [`repos/OWNER/REPO/commits/REF/check-runs`](https://docs.github.com/en/rest/checks/runs?apiVersion=2022-11-28#list-check-runs-for-a-git-reference), is not able to retrieve anything.

`check_name` is filled by Github with the [jobs.<job_id>](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_id) (or [jobs.<job_id>.name](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idname)) and we are filtering it with the [run-name](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#run-name) which never will match :grimacing:

I'm really sad since Github is not puttin any effort to make this run id easily retrievable :disappointed:

Reply:
@LudovicTOURMAN you're definitely right that `job name != run name`. In fact, I have this working on a couple of pipelines but am providing a unique `job name` from within the target `run`. I was just leveraging the existing function but realize this does not align with the readme and would mislead people. That said, I started to refactor for `run name` and noticed you opened a PR of your own. I'm fine to close this one and go with yours. However, I do have some observations and will leave comments. Thanks for the feedback!